runtime/fail_nat.c

```diff
@@ -176,11 +177,8 @@ void caml_array_bound_error(void)
   if (caml_array_bound_error_exn == NULL) {
     caml_array_bound_error_exn =
       caml_named_value("Pervasives.array_bound_error");
-    if (caml_array_bound_error_exn == NULL) {
-      fprintf(stderr, "Fatal error: exception "
-                      "Invalid_argument(\"index out of bounds\")\n");
-      exit(2);
-    }
+    if (caml_array_bound_error_exn == NULL)
```

Comment:
Same thing here: `caml_array_bound_error_exn` can be NULL if the OCaml code hasn't been initialized  yet, and the C code raises the out-of-bounds exception.  

The previous code was careful to report this situation like an uncaught exception `Invalid_argument("index out of bounds")`, and the new code just dumps core.

Reply:
Same argument as above: if we assume that the C code is not calling OCaml runtime functions before initializing that runtime, then this can only happen if the standard library failed to initialize this named value. I consider this as an internal OCaml error.

The assumption that the client code should not call runtime function before initialization is reasonable: you don't want to check in e.g., `caml_alloc` that the runtime is initialized, do you?